Basic-Car-Maintenance/Shared/Odometer/Views/OdometerView.swift

```diff
@@ -12,9 +12,11 @@ struct OdometerView: View {
     @Environment(ActionService.self) var actionService
     
     @State private var viewModel: OdometerViewModel
+    @State private var isShowingEditView = false
+    @State private var selectedOdometerReading: OdometerReading?
 
     init(authenticationViewModel: AuthenticationViewModel) {
-        viewModel = OdometerViewModel(authenticationViewModel: authenticationViewModel)
+        _viewModel = State(initialValue: OdometerViewModel(authenticationViewModel: authenticationViewModel))
```

Comment:
Does this make a difference to write it this way vs the way it was written originally?

Reply:
It gave me an error "Variable 'self.viewModel' used before being initialized" when I added `@State private var selectedOdometerReading: OdometerReading` and I used `_viewModel = State(initialValue:` inside the initialization to resolve the issue. 
Please let me know if you think if it's not the correct way. 
Thank you!

Reply:
That's because you need to give the `selectedOdometerReading` a value

Only defining `@State private var selectedOdometerReading: OdometerReading` does not set a value, and will result in that warning

You need to make `selectedOdometerReading` optional (which you did) and that warning would've have been there. but this initializer doesn't need to change if `selectedOdometerReading` is optional

 Another way is to pass it as an input variable, and then you'd still need something to hold all the different text fields as state properties

Reply:
I gave a value to the selectedOdometerReading  `@State private var selectedOdometerReading = OdometerReading( date: Date(), distance: 100, isMetric: true, vehicle: Vehicle(name: "Example Vehicle", make: "Make", model: "Model"))` but it gave me an error "Cannot convert value of type 'Binding<OdometerReading>' to expected argument type 'Binding<OdometerReading?>'"  on the ```
.sheet(isPresented: $isShowingEditView) {
                        EditOdometerReadingView(vehicles: viewModel.vehicles, selectedOdometerReading: $selectedOdometerReading, viewModel: viewModel)
                    }
```